test/model/autocomplete_test.dart

```diff
@@ -274,35 +276,35 @@ void main() {
     }
   });
 
-  test('MentionAutocompleteView mutating store.users while in progress causes retry', () async {
-    const narrow = CombinedFeedNarrow();
+  test('MentionAutocompleteView mutating store.users while in progress does not '
+      'interrupt the current query results', () async {
+    const narrow = StreamNarrow(1);
     final store = eg.store();
-    for (int i = 0; i < 1500; i++) {
+    for (int i = 0; i < 2500; i++) {
       await store.addUser(eg.user(userId: i, email: 'user$i@example.com', fullName: 'User $i'));
     }
     final view = MentionAutocompleteView.init(store: store, narrow: narrow);
 
     bool done = false;
     view.addListener(() { done = true; });
-    view.query = MentionAutocompleteQuery('User 10000');
+    view.query = MentionAutocompleteQuery('User 110');
 
     await Future(() {});
     check(done).isFalse();
-    await store.addUser(eg.user(userId: 10000, email: 'user10000@example.com', fullName: 'User 10000'));
+    await store.addUser(eg.user(userId: 11000, email: 'user11000@example.com', fullName: 'User 11000'));
     await Future(() {});
     check(done).isFalse();
-    await Future(() {});
-    check(done).isTrue();
-    check(view.results).single
-      .isA<UserMentionAutocompleteResult>()
-      .userId.equals(10000);
-    // new result sticks; no "zombie" result from `store.users` pre-mutation
-    for (int i = 0; i < 10; i++) { // for good measure
+    while (!done) {
       await Future(() {});
```

Comment:
This will become an infinite loop if something goes wrong such that the search never finishes. When a test breaks, it's annoying if the behavior is that it spins forever — it makes CI take much longer to show results (as the test runs until a timeout), and it gets in the way of seeing results from other tests.

That's why it's best to have some bound on the number of iterations. See my suggestion above at https://github.com/zulip/zulip-flutter/pull/693/files#r1632735033 . Or for examples in our existing tests in the repo, see `test/widgets/message_list_test.dart` and search for "for (int "; there's a couple of tests there that do a fling-scroll, and don't want to hard-code an expectation for just how many frames it'll take before the scroll reaches the relevant point, but they do put an upper bound on how many they'll wait for.